api/lib/domain/usecases/generate-username-with-temporary-password.js

```diff
@@ -0,0 +1,49 @@
+const _ = require('lodash');
+const { UserNotAuthorizedToGenerateUsernamePasswordError } = require('../errors');
+
+module.exports = async function generateUsernameWithTemporaryPassword({
+  connectedUserId,
+  schoolingRegistrationId,
+  organizationId,
+  passwordGenerator,
+  encryptionService,
+  userReconciliationService,
+  userRepository,
+  schoolingRegistrationRepository
+}) {
+  const userWithMemberships = await userRepository.getWithMemberships(connectedUserId);
+  _checkIfConnectedUserHasAccessToOrganization(userWithMemberships, organizationId);
+
+  const schoolingRegistration = await schoolingRegistrationRepository.get(schoolingRegistrationId);
+  _checkIfStudentHasAccessToOrganization(schoolingRegistration, organizationId);
+
+  const studentAccount = await userRepository.get(schoolingRegistration.userId);
+  _checkIfstudentAccountHasAlreadyUsername(studentAccount);
+
+  const username = await userReconciliationService.createUsernameByUser({ user: schoolingRegistration , userRepository });
+  const generatedPassword = passwordGenerator.generate();
+  const hashedPassword = await encryptionService.hashPassword(generatedPassword);
+
+  await userRepository.updateUsernameAndPassword(studentAccount.id, username, hashedPassword);
+
+  return { username,generatedPassword };
+};
+
+function _checkIfConnectedUserHasAccessToOrganization(userWithMemberships, organizationId) {
```

Comment:
L'info _Connected_ semble superflue
```suggestion
function _checkIfUserHasAccessToOrganization(userWithMemberships, organizationId) {
```